apps/studio/src/lib/editor/languageData.ts

```diff
@@ -122,3 +125,86 @@ export function getLanguageByContent(content: string): LanguageData | undefined
 export function getLanguageByName(name: string): LanguageData | undefined {
   return Languages.find((lang) => lang.name === name);
 }
+
+type Language = ConnectionType | "sql" | "text" | "html" | "json";
+
+interface CodeMirrorLanguage {
+  mode: string | Record<string, unknown>;
+  hint?: unknown;
+}
+
+export function resolveLanguage(lang: Language): CodeMirrorLanguage {
```

Comment:
This is probably where we can put some of the autocomplete logic that is currently in SQLTextEditor